network/network-address/src/encrypted.rs

```diff
@@ -291,7 +242,7 @@ impl EncNetworkAddress {
         // remove the auth tag suffix, leaving just the decrypted network address
         enc_addr.truncate(auth_tag_offset);
 
-        Ok(RawNetworkAddress::new(enc_addr))
+        Ok(lcs::from_bytes(&enc_addr).unwrap())
```

Comment:
if we assume lcs is infallible we probably should make it not return Result or have a wrapper not return Result but just unwrap inside instead of unwrap at every callsites

Reply:
Humm... good call, this could be fallible actually :/